pkg/vm/vm.go

```diff
@@ -786,10 +786,16 @@ func (v *VM) execute(ctx *Context, op Instruction, parameter []byte) {
 		switch t := arrElem.value.(type) {
 		case *ArrayItem:
```

Comment:
can't we combine these 2 parts of code: 787-790 and 791-795? they look identical

Reply:
Nope, it's Go and even though you can do something like
```go
switch t := arrElem.value.(type) {
		case *ArrayItem, *StructItem:
```
The rest of it just won't work this way. Like compare this: https://play.golang.org/p/FbHgUiPRAsX and this: https://play.golang.org/p/dpAQC5zzDAs

Reply:
They _are_ identical, however we need to be able to access a field in a type-switched structure (`t.value = ...`).